Providers/Scripts/2.4x-2.5x/Scripts/Tests/test_nxProviders.py

```diff
@@ -4824,6 +4827,266 @@ def testSetEnsureAbsentSolutionPurge(self):
         if os.path.exists(nxNPMD.AGENT_RESOURCE_VERSION_PATH):
             self.assertTrue(False, 'agent resource version file exists')
 
+
+class nxOMSAutomationWorkerTestCases(unittest2.TestCase):
+    """
+    Test Case for nxOMSAutomationWorker.py
+    """
+    workspaceId = 'cfd4ef08-4011-428a-8947-0c2f4605980f'
+    agent_id = 'cfd4ef08-4011-428a-8947-0c2f4605980g'
+    AzureDnsAgentSvcZone = 'agentsvc.azure-automation.net'
+    tempWorkingDir = os.path.join(os.getcwd(), 'Scripts/Tests/temp')
+    dummyFileLocation = os.path.join(os.getcwd(), 'Scripts/Tests/dummy_nxOMSAutomationWorker_files')
+    nxOMSAutomationWorker.WORKER_STATE_DIR = tempWorkingDir
+    nxOMSAutomationWorker.WORKER_CONF_FILE_PATH = os.path.join(tempWorkingDir, 'worker.conf')
+    nxOMSAutomationWorker.WORKER_STATE_FILE_PATH = os.path.join(tempWorkingDir, 'state.conf')
+    nxOMSAutomationWorker.REGISTRATION_FILE_PATH = os.path.join(dummyFileLocation, 'register_oms.py')
+    nxOMSAutomationWorker.HYBRID_WORKER_START_PATH = os.path.join(dummyFileLocation, 'main.py')
+    nxOMSAutomationWorker.DSC_RESOURCE_VERSION_FILE = os.path.join(dummyFileLocation, 'VERSION')
+    nxOMSAutomationWorker.LOCAL_LOG_LOCATION = os.path.join(tempWorkingDir, 'nxOMSAutomationWorker.log')
+    nxOMSAutomationWorker.OMS_ADMIN_CONFIG_FILE = os.path.join(dummyFileLocation, 'omsadmin.conf')
+
+    def setUp(self):
+        """
+        Setup Test resources
+        """
+        if not os.path.isdir(self.tempWorkingDir):
+            os.mkdir(self.tempWorkingDir)
+
+    def tearDown(self):
+        """
+        Remove test resoruces
+        """
+        shutil.rmtree(self.tempWorkingDir)
+
+    def find_all_worker_processes(self):
+        retval = []
+        pc = subprocess.Popen(['ps', '-o', 'pid,args', '-C', 'python'], stdout=subprocess.PIPE,
+                              stderr=subprocess.PIPE)
+        result, error = pc.communicate()
+        if pc.returncode != 0:
+            return
+        splitResults = result.split("\n")
+        for s in splitResults:
+            if s.__contains__(self.workspaceId):
+                striped = str.strip(s)
+                try:
+                    i = int(striped.split(' ')[0])
+                    retval.append(i)
+                except:
+                    pass
+        return retval
+
+    def kill_all_workers(self):
+        pids = self.find_all_worker_processes()
+        if pids is None:
+            return
+        for pid in pids:
+            os.kill(pid, signal.SIGTERM)
+
+    def wait_for_worker_to_start(self):
+        for i in range(1, 5, 2):
+            time.sleep(i)
+            pids = self.find_all_worker_processes()
+            if len(pids) > 0:
+                return
+        raise Exception
+
+    def set_automation_worker_state(self, regFilePresent, stateFilePresent, workerLatest, workerRunning):
```

Comment:
What's regFilePresent (worker.conf? if so please rename). Same for workerLatest / workerRunning.